crates/core/src/bc_protocol/floodlight_status.rs

```diff
@@ -0,0 +1,118 @@
+                response_code: 0,
+                stream_type: 0,
+                class: 0x6414,
+            },
+            body: BcBody::ModernMsg(ModernMsg {
+                extension: Some(Extension {
+                    channel_id: Some(self.channel_id),
+                    ..Default::default()
+                }),
+                payload: None,
+            }),
+        };
+
+        sub_get.send(get)?;
+        let msg = sub_get.rx.recv_timeout(RX_TIMEOUT)?;
+        if msg.meta.response_code != 200 {
+            return Err(Error::CameraServiceUnavaliable);
+        }
+
+        if let BcBody::ModernMsg(ModernMsg {
+            payload:
+                Some(BcPayloads::BcXml(BcXml {
+                    floodlight_status_list: Some(floodlight_status_list),
+                    ..
+                })),
+            ..
+        }) = msg.body
+        {
+            Ok(floodlight_status_list)
+        } else {
+            Err(Error::UnintelligibleReply {
+                reply: Box::new(msg),
+                why: "Expected FloodlightStatusList xml but it was not recieved",
+            })
+        }
+    }
+
+    /// Set the floodlight status using the [FloodlightManual] xml
+    pub fn set_floodlight_manual(&self, floodlight_manual: FloodlightManual) -> Result<()> {
+        let connection = self
+            .connection
+            .as_ref()
+            .expect("Must be connected to set floodlight state");
+
+        let msg_num = self.new_message_num();
+        let sub_set = connection.subscribe(msg_num)?;
+
+        let get = Bc {
+            meta: BcMeta {
+                msg_id: MSG_ID_FLOODLIGHT_MANUAL,
+                channel_id: self.channel_id,
+                msg_num,
+                response_code: 0,
+                stream_type: 0,
+                class: 0x6414,
+            },
+            body: BcBody::ModernMsg(ModernMsg {
+                extension: Some(Extension {
+                    channel_id: Some(self.channel_id),
+                    ..Default::default()
+                }),
+                payload: Some(BcPayloads::BcXml(BcXml {
+                    floodlight_manual: Some(floodlight_manual),
+                    ..Default::default()
+                })),
+            }),
+        };
+
+        sub_set.send(get)?;
+        let msg = sub_set.rx.recv_timeout(RX_TIMEOUT)?;
+
+        if let BcMeta {
+            response_code: 200, ..
+        } = msg.meta
+        {
+            Ok(())
+        } else {
+            Err(Error::UnintelligibleReply {
+                reply: Box::new(msg),
+                why: "The camera did not accept the Floodlight manual state",
+            })
+        }
+    }
+
+    /// Set the floodlight status for a duration
+    pub fn floodlight_light_set(&self, state: bool, duration: u16) -> Result<()> {
+        let mut floodlight_state = self.get_floodlight_status()?;
+        for floodlight in floodlight_state.floodlight_status_list.iter_mut() {
```

Comment:
I think just `.iter()` is ok here as it dosen't need to be mutable. You can drop the `mut` too then on the status